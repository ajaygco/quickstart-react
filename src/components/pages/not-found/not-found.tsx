// Modules
import * as React from "react";
import { Link } from "react-router-dom";

// Shared
import { Logo } from "@app/components/shared/logo.shared";

// Types
type NotFoundProps = {};

// Component
export const NotFound: React.FC<NotFoundProps> = (): React.ReactElement => {
  return (
    <div className="w-full h-full flex items-center justify-center">
      <div className="w-[460px] m-auto flex flex-col items-center p-6">
        <div className="animate__animated animate__bounceIn mb-2">
          <span className="text-2xl text-gray-800 font-display font-bold">
            Page Not Found
          </span>
        </div>

        <div className="animate__animated animate__fadeInUp animate__delay-1s text-center mb-8">
          <span className="text-gray-600">
            The page you are looking for does not exist. You may have followed a
            broken link or the page might have been moved or deleted.
          </span>
        </div>

        <div className="animate__animated animate__fadeInUp animate__delay-2s">
          <Link to="/">
            <span className="text-4xl text-gray-800">
              <Logo />
            </span>
          </Link>
        </div>
      </div>
    </div>
  );
};
