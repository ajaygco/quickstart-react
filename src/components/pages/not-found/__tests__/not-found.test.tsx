// bun run test src/components/pages/not-found/__tests__/not-found.test.tsx

import * as React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import { NotFound } from "@app/components/pages/not-found/not-found";

describe("NotFound", () => {
  it("should render correctly", () => {
    const wrapper = render(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
