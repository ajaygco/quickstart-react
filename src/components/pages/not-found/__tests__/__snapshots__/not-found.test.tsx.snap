// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-full h-full flex items-center justify-center"
      >
        <div
          class="w-[460px] m-auto flex flex-col items-center p-6"
        >
          <div
            class="animate__animated animate__bounceIn mb-2"
          >
            <span
              class="text-2xl text-gray-800 font-display font-bold"
            >
              Page Not Found
            </span>
          </div>
          <div
            class="animate__animated animate__fadeInUp animate__delay-1s text-center mb-8"
          >
            <span
              class="text-gray-600"
            >
              The page you are looking for does not exist. You may have followed a broken link or the page might have been moved or deleted.
            </span>
          </div>
          <div
            class="animate__animated animate__fadeInUp animate__delay-2s"
          >
            <a
              href="/"
            >
              <span
                class="text-4xl text-gray-800"
              >
                <span
                  class="font-brand lowercase"
                >
                  Kickstart
                </span>
              </span>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-full h-full flex items-center justify-center"
    >
      <div
        class="w-[460px] m-auto flex flex-col items-center p-6"
      >
        <div
          class="animate__animated animate__bounceIn mb-2"
        >
          <span
            class="text-2xl text-gray-800 font-display font-bold"
          >
            Page Not Found
          </span>
        </div>
        <div
          class="animate__animated animate__fadeInUp animate__delay-1s text-center mb-8"
        >
          <span
            class="text-gray-600"
          >
            The page you are looking for does not exist. You may have followed a broken link or the page might have been moved or deleted.
          </span>
        </div>
        <div
          class="animate__animated animate__fadeInUp animate__delay-2s"
        >
          <a
            href="/"
          >
            <span
              class="text-4xl text-gray-800"
            >
              <span
                class="font-brand lowercase"
              >
                Kickstart
              </span>
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
